pubnub-kotlin/pubnub-kotlin-impl/src/main/kotlin/com/pubnub/internal/managers/TokenParser.kt

```
@@ -16,6 +16,67 @@ import kotlin.collections.set
+    /**
+     * Finds all tokens in a list that match a given version.
+     */
+    fun findTokensByVersion(tokens: List<String>, version: Int): List<PNToken> {
+        val result = mutableListOf<PNToken>()
+        for (token in tokens) {
+            try {
+                val parsed = unwrapToken(token)
+                if (parsed.version == version) {
+                    result.add(parsed)
+                }
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)